src/zenml/integrations/kubeflow/flavors/kubeflow_orchestrator_flavor.py

```diff
@@ -116,6 +121,37 @@ def _migrate_pod_settings(cls, values: Dict[str, Any]) -> Dict[str, Any]:
             values["node_affinity"] = {}
             values["node_selectors"] = {}
 
+        client_creds_error = AssertionError(
+            "`client_username` and `client_password` both need to be set."
+        )
+        if (
+            values.get("client_username")
+            and values.get("client_password") is None
+        ):
+            raise client_creds_error
+        if (
+            values.get("client_password")
+            and values.get("client_username") is None
+        ):
+            raise client_creds_error
+
+        if values.get("kubeflow_hostname"):
+            if "/pipeline" not in values.get("kubeflow_hostname"):
+                logger.warning(
+                    "It looks like you have specified a `kubeflow_hostname` but it doesn't seem to "
+                    "end with `/pipeline`, which is not usual. Please ensure this is intended."
+                )
+
+            if (
+                values.get("client_username") is None
+                and values.get("client_password") is None
+                and values.get("client_args") is None
+            ):
+                logger.warning(
```

Comment:
This assumes that setting a kubeflow hostname necessarily means multi-tenant deployment. Is that actually the case?

Reply:
Yes this is why its just a warning and it says "You see to be...". However, I have changed the error message now to make it not tied to multi-tenant